tests/nvme/001

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+#
+# Regression test for commit 3d030e41d96f ("nvme: add tracepoint for
+# nvme_setup_cmd")
+#
+# Copyright (C) 2018 Johannes Thumshirn
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+DESCRIPTION="enable nvme_setup_nvm_cmd trace point and submit I/O to the
+device"
+
+QUICK=1
+
+# This function normalizes the variable parts of the trace on the individual
+# test system so the known good result in test/nvme/001.out matches if all is
+# OK.
+# If anyone finds a readable way of definig this filter I'm all ears.
+filter_trace() {
+	sed -r "s/dd-[0-9]+\s+\[[0-9]+\]\s+\.{4}\s+[0-9]+\.[0-9]+: nvme_setup_nvm_cmd:\s+qid=[0-9]+, nsid=[0-9]+, cmdid=[0-9]+, flags=0x0, meta=0x0, cmd=\(nvme_cmd_read slba=0, len=31, ctrl=0x0, dsmgmt=0, reftag=0\)/dd-XXX   [XXX] ....     X.XXXXXX: nvme_setup_nvm_cmd: qid=X, nsid=X, cmdid=X, flags=0x0, meta=0x0, cmd=(nvme_cmd_read slba=0, len=31, ctrl=0x0, dsmgmt=0, reftag=0)/"
+}
+
+test_device() {
+	echo "Running ${TEST_NAME}"
+	cleanup_debugfs=0
+	cleanup_tracefs=0
+
+	if ! grep -q debugfs /proc/mounts; then
+		mount -t debugfs none /sys/kernel/debug
+		cleanup_debugfs=1
+	fi
+
+	if ! grep -q tracefs /proc/mounts; then
+		mount -t tracefs none /sys/kernel/debug/tracing
+		cleanup_tracefs=1
+	fi
+
+	echo 1 > /sys/kernel/debug/tracing/events/nvme/nvme_setup_nvm_cmd/enable
+
+	dd if=${TEST_DEV} of=/dev/null bs=512 count=1 >>$FULL 2> /dev/null
```

Comment:
Oh, one other thing, this needs `iflag=direct` so that we make sure it goes to the device, right?

Reply:
Wasn't needed in my tests but better save than sorry I guess.